### PR TITLE
docs: update faq about auto-complete

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [X] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  update faq about auto-complete  |
| 🇨🇳 Chinese |  更新自动完成组件 FAQ  |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at a55bd3d</samp>

This pull request updates the documentation of the `AutoComplete` component in both English and Chinese. It adds a new FAQ subsection that explains the logic behind the dropdown menu display and how to control it with props.

### <samp>🤖 Generated by Copilot at a55bd3d</samp>

*  Add a new FAQ subsection to explain the conditions for the AutoComplete component dropdown menu to be displayed ([link](https://github.com/ant-design/ant-design/pull/44082/files?diff=unified&w=0#diff-f04508ef988a95e022e0488be2b02506610901732efa4dfba77f7ac6e1ae65d1R89-R97), [link](https://github.com/ant-design/ant-design/pull/44082/files?diff=unified&w=0#diff-54fefadfae27caa4550cb59eb92b84d4a0ec3b9b17bb56f9ec3cae2948886db3R91-R99))

### CODE REVIEW

File: components/auto-complete/index.zh-CN.md [hunk 1]
```diff
@@ -88,3 +88,12 @@ demo:
...
 相关 issue：[#18230](https://github.com/ant-design/ant-design/issues/18230) [#17916](https://github.com/ant-design/ant-design/issues/17916)
+
+### 为何设置了 `open` 或 `defaultOpen` 时无法自动展开下拉菜单？
+
+在 AutoComplete 中，面板是否展示依赖于两个方面：
+
+1. `open` 或 `defaultOpen` 的值为 `true`
+2. `options` 的值为非空数组
+
+如果 `options` 为空数组，那么无论 `open` 或 `defaultOpen` 的值为何，面板都不会自动展示。
```
Comment (afc163) [inline components/auto-complete/index.zh-CN.md:99]: 这种情况是不是给一个 empty 节点更好？

Comment (zombieJ) [reply components/auto-complete/index.zh-CN.md:99]: AutoComplete 本质是 Input，空 `options` 的时候显示 empty 会让人误以为不能操作。其实是可以输入的。

Comment (kiner-tang) [reply components/auto-complete/index.zh-CN.md:99]: 不过这是手动传入 open 或 defaultOpen 打开的，感觉这个时候展示 empty 也没问题

Comment (afc163) [reply components/auto-complete/index.zh-CN.md:99]: > AutoComplete 本质是 Input，空 options 的时候显示 empty 会让人误以为不能操作。其实是可以输入的。
也对，options 可能是输入之后异步读来的。

File: components/auto-complete/index.zh-CN.md [hunk 2]
```diff
@@ -88,3 +88,12 @@ demo:
 请使用 `onChange` 进行受控管理。`onSearch` 触发于搜索输入，与 `onChange` 时机不同。此外，点击选项时也不会触发 `onSearch` 事件。
 
 相关 issue：[#18230](https://github.com/ant-design/ant-design/issues/18230) [#17916](https://github.com/ant-design/ant-design/issues/17916)
+
+### 为何设置了 `open` 或 `defaultOpen` 时无法自动展开下拉菜单？
```
Comment (zombieJ) [inline components/auto-complete/index.zh-CN.md:92]: 为何 `options` 为空时，受控 `open` 展开不会显示下拉菜单？

File: components/auto-complete/index.zh-CN.md [hunk 3]
```diff
@@ -88,3 +88,12 @@ demo:
...
+
+在 AutoComplete 中，面板是否展示依赖于两个方面：
```
Comment (zombieJ) [inline components/auto-complete/index.zh-CN.md:94]: 这里的描述只讲了行为，没讲为什么。把我的回复抄一下哈~

File: components/auto-complete/index.zh-CN.md [hunk 4]
```diff
@@ -89,11 +89,6 @@ demo:
...
-### 为何设置了 `open` 或 `defaultOpen` 时无法自动展开下拉菜单？
+### 为何 options 为空时，受控 open 展开不会显示下拉菜单？
 
-在 AutoComplete 中，面板是否展示依赖于两个方面：
-
-1. `open` 或 `defaultOpen` 的值为 `true`
-2. `options` 的值为非空数组
-
-如果 `options` 为空数组，那么无论 `open` 或 `defaultOpen` 的值为何，面板都不会自动展示。
+AutoComplete 组件本质上是 Input 输入框的一种扩展，当 options 为空时，显示空文本会让用户误以为该组件不可操作，实际上它仍然可以进行文本输入操作。因此，为了避免给用户带来困惑，当 options 为空时，open 属性为 true 也不会展示下拉菜单，需要与 options 属性配合使用。
```
Comment (afc163) [inline components/auto-complete/index.zh-CN.md:94]:
```suggestion
AutoComplete 组件本质上是 Input 输入框的一种扩展，当 `options` 为空时，显示空文本会让用户误以为该组件不可操作，实际上它仍然可以进行文本输入操作。因此，为了避免给用户带来困惑，当 `options` 为空时，`open` 属性为 `true` 也不会展示下拉菜单，需要与 `options` 属性配合使用。
```

**Review by zombieJ — APPROVED** [review]

**Pull request merged**